frontend/src/components/user/SavedSearchQueryUserControl.tsx

```diff
@@ -0,0 +1,32 @@
+"use client";
+
+import { environment } from "src/constants/environments";
+
+import { useTranslations } from "next-intl";
+import { usePathname, useSearchParams } from "next/navigation";
+
+import SavedSearchQuery from "src/components/search/SavedSearchQuery";
+
+export const SearchQuerySaveUserControl = () => {
+  const path = usePathname();
+  const searchParams = useSearchParams();
+  const query = searchParams?.toString() ? `?${searchParams.toString()}` : "";
+  const url = `${environment.NEXT_PUBLIC_BASE_URL}${path}${query}`;
```

Comment:
This works in production mode locally, and should be correctly set in dev, but not 100% certain. Can update if not.

Reply:
I'd lean on using the Next functions first, but we could fall back on that.

Reply:
yeah, I'm fine with this as is but if it causes problems in dev I think using window.location is a good alternative